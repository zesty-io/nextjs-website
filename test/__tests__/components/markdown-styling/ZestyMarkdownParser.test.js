// MDH3.test.js

import React from 'react';
import { render } from '@testing-library/react';
import {
  MDH3,
  MDH1,
  MDH2,
  MDParagraph,
  MDCodeBlock,
  MDImage,
  MDBlockquote,
  MDStrong,
  MDItalic,
} from '../../../../src/components/markdown-styling/components';

jest.mock('uuid', () => ({ v4: () => '123456789' }));
describe('MDH3', () => {
  it('should render the MDH3 component with correct props', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'text',
          value: 'Heading 3 Text',
        },
      ],
    };

    // Render the MDH3 component with the sample node data
    const { getByText, getByRole, getByTestId } = render(<MDH3 node={node} />);

    const item = getByTestId('mdh3-container');
    expect(item).toBeDefined();
    expect(item).toMatchSnapshot();
    // Expect the Typography component to be rendered with the correct variant and component
    const typographyElement = getByText('Heading 3 Text');
    expect(typographyElement).toBeInTheDocument();
    expect(typographyElement.tagName.toLowerCase()).toBe('h3');
    expect(typographyElement).toHaveClass('MuiTypography-h6');

    // Expect the Typography component to have the correct id attribute
    const expectedId = 'heading-3-text';
    expect(typographyElement).toHaveAttribute('id', expectedId);
  });
});

describe('MDH1', () => {
  it('should render the MDH1 component with correct props and styling', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'text',
          value: 'Heading 1 Text',
        },
      ],
    };

    const { getByText, getByRole, getByTestId } = render(<MDH1 node={node} />);

    const boxElement = getByTestId('box-container');
    const item = getByTestId('box-container');
    expect(item).toBeDefined();
    expect(item).toMatchSnapshot();
    expect(boxElement).toBeInTheDocument();
    expect(boxElement).toMatchSnapshot();
    const typographyElement = getByText('Heading 1 Text');
    expect(typographyElement).toBeInTheDocument();
    expect(typographyElement.tagName.toLowerCase()).toBe('h1');
    const linkElement = getByRole('link');
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', '#heading-1-text');
  });
});

describe('MDH2', () => {
  it('should render the MDH2 component with correct props and styling', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'text',
          value: 'Heading 2 Text',
        },
      ],
    };

    // Render the MDH2 component with the sample node data
    const { getByText, getByRole, getByTestId } = render(<MDH2 node={node} />);
    const item = getByTestId('mdh2-container');
    expect(item).toBeDefined();
    expect(item).toMatchSnapshot();

    // Expect the Box component to be rendered with the correct styling
    const boxElement = getByRole('heading', { level: 2 });
    expect(boxElement).toBeInTheDocument();

    // Expect the Typography component to be rendered with the correct content, props, and styling
    const typographyElement = getByText('Heading 2 Text');
    expect(typographyElement).toBeInTheDocument();
    expect(typographyElement.tagName.toLowerCase()).toBe('h2');

    // Expect the Link component to be rendered with the correct props and styling
    const linkElement = getByRole('link');
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', '#heading-2-text');
  });
});

describe('MDParagraph', () => {
  it('should render the MDParagraph component with correct content and links', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'text',
          value: 'This is a paragraph with a ',
        },
        {
          type: 'a',
          props: {
            href: 'https://example.com',
            children: [
              {
                type: 'text',
                value: 'link',
              },
            ],
          },
        },
        {
          type: 'text',
          value: ' and some keywords like Apple and Banana.',
        },
      ],
    };

    // Sample mainKeywords and productGlossary for testing
    const mainKeywords = ['apple', 'banana'];
    const productGlossary = [
      {
        target_words: ['apple'],
        url: 'https://apple-url.com',
        description: 'This is an apple.',
      },
      {
        target_words: ['banana'],
        url: 'https://banana-url.com',
        description: 'This is a banana.',
      },
    ];

    const children = [
      'By using URL parameters, businesses can craft dyna… possibilities for enhancing the user experience.',
    ];
    // Render the MDParagraph component with the sample node data and props
    const { container, getByTestId } = render(
      <MDParagraph
        node={node}
        mainKeywords={mainKeywords}
        productGlossary={productGlossary}
        children={children}
      />,
    );

    const item = getByTestId('box-container');
    expect(item).toBeDefined();
    expect(item).toMatchSnapshot();
    // Expect the rendered output to contain the correct content and links
    expect(container.innerHTML).toContain(
      `<p data-testid=\"box-container\">By using URL parameters, businesses can craft dyna… possibilities for enhancing the user experience.</p>`,
    );
  });

  // Add more test cases as needed to cover other scenarios and edge cases.
});

describe('MDCodeBlock', () => {
  it('should render the MDCodeBlock component with correct value and styles', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'code',
          children: [
            {
              type: 'text',
              value:
                "const greeting = 'Hello, world!';\nconsole.log(greeting);",
            },
          ],
        },
      ],
    };

    // Render the MDCodeBlock component with the sample node data
    const { getByText, getByTestId } = render(<MDCodeBlock node={node} />);

    // Expect the CodeMirror component to receive the correct value prop
    const codeMirror = getByTestId('code-mirror');
    expect(codeMirror).toBeInTheDocument();

    // Expect the Box component to be rendered with the correct padding
  });
});

describe('MDImage', () => {
  it('should render MDLargeImage if node has title', () => {
    const node = {
      properties: {
        src: 'image-large.jpg',
        alt: 'Large Image',
        title: 'Large Image Title',
      },
    };

    const { getByTestId } = render(<MDImage node={node} />);
    const largeImage = getByTestId('md-large-image');

    expect(largeImage).toBeInTheDocument();
  });

  it('should render MDSmallImage if node does not have title', () => {
    const node = {
      properties: {
        src: 'image-small.jpg',
        alt: 'Small Image',
      },
    };

    const { getByTestId } = render(<MDImage node={node} />);
    const smallImage = getByTestId('md-small-image');

    expect(smallImage).toBeInTheDocument();

    expect(smallImage).toMatchSnapshot();
  });
});

describe('MDBlockquote', () => {
  it('should render the MDBlockquote component with correct content and styles', () => {
    // Sample node data for testing
    const node = {
      children: [
        {
          type: 'blockquote',
          children: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  value: 'This is a blockquote text.',
                },
              ],
            },
          ],
        },
      ],
    };

    // Render the MDBlockquote component with the sample node data
    const { getByText, getByTestId } = render(<MDBlockquote node={node} />);

    // Expect the Box component to be rendered with the correct styling
    const boxElement = getByTestId('blockquote-box');
    expect(boxElement).toBeInTheDocument();

    expect(boxElement).toMatchSnapshot();
  });
});

describe('MDStrong', () => {
  it('renders the text with the correct fontWeight', () => {
    const mockNode = {
      children: [
        {
          children: [
            {
              value: 'Hello, World!',
            },
          ],
        },
      ],
    };
    const { getByText, getByTestId } = render(<MDStrong node={mockNode} />);
    const strongText = getByText('Hello, World!');
    expect(strongText).toBeInTheDocument();
    expect(strongText).toHaveStyle('font-weight: 700');

    const boxelement = getByTestId('MDStrong-component');
    expect(boxelement).toBeInTheDocument();

    expect(boxelement).toMatchSnapshot();
  });
});

describe('MDItalic', () => {
  it('renders the text with the correct fontStyle and fontWeight', () => {
    const mockNode = {
      children: [
        {
          children: [
            {
              value: 'Hello, World!',
            },
          ],
        },
      ],
    };
    const { getByText, getByTestId } = render(<MDItalic node={mockNode} />);
    const italicText = getByText('Hello, World!');
    expect(italicText).toBeInTheDocument();
    expect(italicText).toHaveStyle('font-style: italic');
    expect(italicText).toHaveStyle('font-weight: 400');

    const boxelement = getByTestId('MDItalic-component');
    expect(boxelement).toBeInTheDocument();

    expect(boxelement).toMatchSnapshot();
  });
});
