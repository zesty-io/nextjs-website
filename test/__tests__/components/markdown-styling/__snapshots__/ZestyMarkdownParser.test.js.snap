// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDBlockquote should render the MDBlockquote component with correct content and styles 1`] = `
<div
  class="MuiBox-root css-hwyg59"
  data-testid="blockquote-box"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`MDH1 should render the MDH1 component with correct props and styling 1`] = `
<div
  class="MuiBox-root css-z6qf5z"
  data-testid="box-container"
>
  <h1
    class="MuiTypography-root MuiTypography-h4 css-9o07c9-MuiTypography-root"
    id="heading-1-text"
  >
    Heading 1 Text
  </h1>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ka6tax-MuiTypography-root-MuiLink-root"
    href="#heading-1-text"
  >
    #
  </a>
</div>
`;

exports[`MDH1 should render the MDH1 component with correct props and styling 2`] = `
<div
  class="MuiBox-root css-z6qf5z"
  data-testid="box-container"
>
  <h1
    class="MuiTypography-root MuiTypography-h4 css-9o07c9-MuiTypography-root"
    id="heading-1-text"
  >
    Heading 1 Text
  </h1>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ka6tax-MuiTypography-root-MuiLink-root"
    href="#heading-1-text"
  >
    #
  </a>
</div>
`;

exports[`MDH2 should render the MDH2 component with correct props and styling 1`] = `
<div
  class="MuiBox-root css-1ouodky"
  data-testid="mdh2-container"
>
  <h2
    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    id="heading-2-text"
  >
    Heading 2 Text
  </h2>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-om9vyh-MuiTypography-root-MuiLink-root"
    href="#heading-2-text"
  >
    #
  </a>
</div>
`;

exports[`MDH3 should render the MDH3 component with correct props 1`] = `
<h3
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  data-testid="mdh3-container"
  id="heading-3-text"
>
  Heading 3 Text
</h3>
`;

exports[`MDImage should render MDSmallImage if node does not have title 1`] = `
<div
  class="MuiBox-root css-1ap5130"
  data-testid="md-small-image"
>
  <div
    class="MuiStack-root css-o6cat4-MuiStack-root"
  >
    <div>
      <img
        alt="Small Image"
        src="image-small.jpg"
        style="cursor: pointer; max-width: 100%; max-height: 100%;"
      />
    </div>
  </div>
</div>
`;

exports[`MDItalic renders the text with the correct fontStyle and fontWeight 1`] = `
<div
  class="MuiStack-root css-nen11g-MuiStack-root"
  data-testid="MDItalic-component"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-c83ayk-MuiTypography-root"
  >
    Hello, World!
  </p>
</div>
`;

exports[`MDParagraph should render the MDParagraph component with correct content and links 1`] = `
<p
  data-testid="box-container"
>
  By using URL parameters, businesses can craft dynaâ€¦ possibilities for enhancing the user experience.
</p>
`;

exports[`MDStrong renders the text with the correct fontWeight 1`] = `
<div
  class="MuiStack-root css-nen11g-MuiStack-root"
  data-testid="MDStrong-component"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
  >
    Hello, World!
  </p>
</div>
`;
